src/ssl/support.cc

```diff
@@ -748,6 +786,8 @@ Ssl::Initialize(void)
 
     SQUID_OPENSSL_init_ssl();
 
+    CRYPTO_set_mem_functions(CryptoMalloc, CryptoRealloc, CryptoFree);
```

Comment:
BTW, OpenSSL has its own statistics for allocations, available via the `CRYPTO_get_alloc_counts(int *mcount, int *rcount, int *fcount)` call.

Reply:
... but only "If the library is built with the crypto-mdebug option". I assume that popular Ubuntu OpenSSL packages do not enable that library option. Is that assumption wrong?

Reply:
I checked that OpenSSL (1.1.1 and 13.0.13) goes without crypto-mdebug in Ubuntu 18.04, 20.04, 24.04.

Reply:
Thank you. Your research confirms that we should not rely on CRYPTO_get_alloc_counts(), so this PR is on the right track with adding custom stats.